src/Components/Home.js

```diff
@@ -0,0 +1,256 @@
+
+  const countries = useSelector((state) => state.countries);
+
+  const handleFilter = (e) => {
+    const { value } = e.target;
+    setState({ ...State, filter: value });
+  };
+
+  const openPopup = (e) => {
+    if (e.target.hasAttribute('alt')) {
+      setState({ ...State, popup: true, keyId: e.target.parentElement.id });
+    }
+  };
+
+  if (State.popup) {
+    document.body.style.overflow = 'hidden';
+  } else {
+    document.body.style.overflow = 'auto';
+  }
+
+  const closePopup = () => {
+    setState({ ...State, popup: false });
+  };
+
+  const handleSearch = (e) => {
+    e.preventDefault();
+    const { value } = e.target;
+    let newValue;
+    if (value) {
+      newValue = value.trim().toLowerCase();
+    } else {
+      newValue = '';
+    }
+    setState(
+      { ...State, search: newValue },
+    );
+  };
+
+  const toggleVisible = () => {
+    const scrolled = document.documentElement.scrollTop;
+    if (scrolled > 1000) {
+      setVisible(true);
+    } else if (scrolled <= 1000) {
+      setVisible(false);
+    }
+  };
+
+  window.addEventListener('scroll', toggleVisible);
+
+  const scrollToTop = () => {
+    window.scrollTo({
+      top: 0,
+      behavior: 'smooth',
+    });
+  };
+
+  return (
+    <section className={theme === 'default' ? 'home home-df' : 'home'}>
+      <Popup trigger={State.popup} keyId={State.keyId} closePopup={closePopup} theme={theme} />
+      <div className="container">
+        <form onSubmit={(e) => e.preventDefault()}>
+          <div className="SearchBar-container" onChange={handleSearch} onBlur={handleSearch}>
+            <input type="search" placeholder="Search for a country" />
+          </div>
+          <div className="filter-container">
+            <select name="Regions" onChange={handleFilter}>
+              <option value="All">Filter by Region</option>
+              <option value="Africa">Africa</option>
+              <option value="Americas">Americas</option>
+              <option value="Asia">Asia</option>
+              <option value="Europe">Europe</option>
+              <option value="Oceania">Oceania</option>
+            </select>
+          </div>
+        </form>
+        <button
+          type="button"
+          className={theme === 'default' ? 'scroll-up' : 'scroll-up scroll-up-white'}
+          style={{ display: visible ? 'inline' : 'none' }}
+          onClick={scrollToTop}
+          onKeyDown={scrollToTop}
+        >
+          &#8593;
+        </button>
+        <div className="cards-container">
+          {countries.map((country) => { // eslint-disable-line
+            if (State.filter === country.region) {
+              if (State.search === '') {
+                return (
+                  <div
+                    className={theme === 'default' ? 'card card-df' : 'card'}
+                    key={country.id}
+                    id={country.id}
+                    onClick={openPopup}
+                    onKeyDown={openPopup}
```

Comment:
- [ ] You made great work but there is one issue in which you use open and close popup the goal is to use react-router-dom to make sure, you understand it well. And the detail page for each country has its unique route within the SPA as stated in the requirements. You can refer for [this tutorial](https://www.youtube.com/watch?v=UWYOC8g5N_0&list=PLC3y8-rFHvwjkxt8TOteFdT_YmzwpBlrG) for good explanation for react-router-dom.
![image](https://user-images.githubusercontent.com/53402600/182872230-bf372501-1091-4177-a838-a8ad4d5fb97a.png)